### DISCUSSION TITLE
Can one use the sonar-openedge plugin when not build with Ant or PCT

### DISCUSSION THREAD

**Discussion by jeffcruz4** [Riverside-Software/sonar-openedge#1014] [Q&A]
Based on this (https://github.com/Riverside-Software/sonar-openedge/wiki/Getting-started#build-process-using-ant-and-pct) we can only use the sonar-openedge plugin when not build with Ant or PCT.  Is that correct?

**Comment by gquerret** [comment]
The documentation says:
> The OpenEdge plugins only work if you build your applications using Ant and PCT, as they expect a specific directory structure for the build directories.

The "only works if" part is not exactly true; you can analyze your source code without using Ant / PCT, but you'll lack some information:
* compiler warnings won't be imported
* XREF based rules won't be executed, as the SonarQube scanner expects files to be in a specific location
* Code coverage won't work if the rcode can't be associated with the source code

Apart from that, you can use your own build tool.

**Reply by jeffcruz4** [reply]
Understood. Thank you.  We have successfully run a scan but are getting 18k parsing errors with the below error message: 
"No viable alternative at input 'address' in /data/workspace/Branch-Lakewood_snagara7-patch-4/inc/nameconst.i:139:39"

Any chance you all have seen this before?

**Reply by gquerret** [reply]
The message means that the parser failed at line 139 (char 39) of include file nameconst.i. Including the line (or block of lines) here might help. Did you setup all [properties](https://github.com/Riverside-Software/sonar-openedge/wiki/Getting-started#analysis-with-sonarqube-scanner) correctly ? Especially `sonar.oe.databases` which should point to the DF files of your databases.
